web/src/api/getZone.ts

```diff
@@ -35,17 +45,27 @@ const getZone = async (
   throw new Error(await response.text());
 };
 
-// TODO: The frontend (graphs) expects that the datetimes in state are the same as in zone
-// should we add a check for this?
 const useGetZone = (): UseQueryResult<ZoneDetails> => {
-  const { zoneId } = useParams<RouteParameters>();
-  const { urlTimeAverage } = useParams<RouteParameters>();
+  const { zoneId, urlTimeAverage, urlDatetime } = useParams<RouteParameters>();
+
   const timeAverage = urlTimeAverage
     ? URL_TO_TIME_AVERAGE[urlTimeAverage]
     : TimeAverages.HOURLY;
   return useQuery<ZoneDetails>({
-    queryKey: [QUERY_KEYS.ZONE, { zone: zoneId, aggregate: timeAverage }],
-    queryFn: async () => getZone(timeAverage, zoneId),
+    queryKey: [
+      QUERY_KEYS.ZONE,
+      {
+        zone: zoneId,
+        aggregate: urlTimeAverage,
+        targetDatetime: urlDatetime,
+      },
+    ],
+    queryFn: async () => {
+      if (!zoneId) {
```

Comment:
Maybe we want to redirect the page instead? Or just use the presence of zoneId to enable/disable the query if we'd prefer to silently fail